src/components/RulesEngine/utils/generateEventHash.js

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import fnv1a32Hex from "../../../utils/fnv1a32Hex.js";
+
+/**
+ * Generates an event history hash from an object by removing keys with empty values, sorting the keys,
+ * converting the object to a string, and hashing the string.
+ * @param {Object} o - The object to process.
+ * @returns {string} - The hash of the processed object.
+ */
+export default (o) => {
+  const obj = structuredClone(o);
+
+  const objectString = Object.keys(obj)
+    .sort()
+    .reduce((result, key) => {
+      const value = obj[key];
+
+      // eslint-disable-next-line eqeqeq
+      if (value == undefined || value === "") {
```

Comment:
This seems to be just a top-level hashing i.e. it does not descend into nested objects. Is that any concern?

Reply:
Not at this moment. The objects that this method will receive are 1 level depth with string keys and values.

I asked the AJO mobile team how do they serialize objects or arrays with higher depth and they couldn’t provide an answer.